kernel/mount.c

```diff
@@ -299,6 +310,114 @@ static const char* _find_arg(const char* args[], const char* name)
 }
 #endif /* MYST_ENABLE_EXT2FS */
 
+#ifdef MYST_ENABLE_HOSTFS
+static bool _find_mount_source(const char* source, char path[PATH_MAX])
+{
+    bool found = false;
+    for (size_t i = 0; i < _mount_table_size; i++)
+    {
+        if (strcmp(_mount_table[i].source, source) == 0)
+        {
+            found = true;
+            myst_strlcpy(path, _mount_table[i].path, PATH_MAX);
+            break;
+        }
+    }
+    return found;
+}
+
+static long _new_automount_dir(char mountdir[PATH_MAX])
+{
+    long ret = -1;
+    int mount_num = 1;
+
+    myst_strlcpy(mountdir, AUTOMOUNT_DIR, PATH_MAX);
+    if (!_created_automount_dir)
+    {
+        ECHECK(myst_syscall_mkdir(mountdir, 0x777));
+        _created_automount_dir = true;
+    }
+
+    while (1)
+    {
+        snprintf(mountdir, PATH_MAX, "%s/%d", AUTOMOUNT_DIR, mount_num++);
+        if (myst_syscall_access(mountdir, F_OK) != 0)
+        {
+            ECHECK(myst_syscall_mkdir(mountdir, 0x777));
+            break;
+        }
+    }
+
+    ret = 0;
+
+done:
+
+    if (ret != 0)
+        mountdir[0] = '\0';
+
+    return ret;
+}
+
+static int _mount_single_file(
+    myst_fs_t* fs,
+    const char* source,
+    const char* target)
+{
+    int ret = -1;
+    struct locals
+    {
+        char sourcedir[PATH_MAX];
+        char sourcebase[PATH_MAX];
+        char mountpath[PATH_MAX];
+    };
+    struct locals* locals = NULL;
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    ECHECK(myst_split_path(
+        source, locals->sourcedir, PATH_MAX, locals->sourcebase, PATH_MAX));
+
+    if (!_find_mount_source(locals->sourcedir, locals->mountpath))
+    {
+        ECHECK(_new_automount_dir(locals->mountpath));
+        ECHECK(myst_mount(fs, locals->sourcedir, locals->mountpath));
+    }
+
+    /* create a symlink */
+    if (myst_syscall_access(target, F_OK) == 0 &&
+        myst_syscall_unlink(target) != 0)
+    {
+        myst_eprintf("kernel: unlink failed: %s\n", target);
+        ERAISE(-EINVAL);
+    }
+
+    myst_strlcpy(locals->mountpath + strlen(locals->mountpath), "/", PATH_MAX);
```

Comment:
You can just use ``myst_strcat`` like this:

```
myst_strlcat(locals->mountpath, "/", PATH_MAX);
```

Reply:
Done.